megamek/src/megamek/common/weapons/ArtilleryWeaponIndirectFireHandler.java

```diff
@@ -152,7 +152,9 @@ public boolean handle(IGame.Phase phase, Vector<Report> vPhaseReport) {
             System.err.println("Artillery Entity is null!");
             return true;
         }
-        Mounted ammoUsed = ae.getEquipment(aaa.getAmmoId());
+        //Trailers can share ammo, which means the entity carrying the ammo might
+        Entity ammoCarrier = game.getEntity(aaa.getAmmoCarrier());
```

Comment:
Could this return `null` if `aaa.getAmmoCarrier()` returns `Entity.NONE`? Or can that never happen?

Reply:
That can't happen. AmmoCarrier will return the firing entity 95% of the time.